### PR TITLE
refactor(VRadioGroup/VRadio): convert to ts

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Converted `v-radio-group` and `v-radio` to typescript. Converted tests from avoriaz to vue-test-utils and typescript.

## Motivation and Context
Normalize `v-radio-group` behavior with `v-item-group`

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
<template>
  <v-container fluid>
    <v-radio-group
      v-model="column"
      column
    >
      <v-radio
        label="Option 1"
        value="radio-1"
      ></v-radio>
      <v-radio
        label="Option 2"
        value="radio-2"
      ></v-radio>
    </v-radio-group>
    <hr>
    <v-radio-group
      v-model="row"
      row
    >
      <v-radio
        label="Option 1"
        value="radio-1"
      ></v-radio>
      <v-radio
        label="Option 2"
        value="radio-2"
      ></v-radio>
    </v-radio-group>
  </v-container>
</template>

<script>
  export default {
    data () {
      return {
        column: null,
        row: null
      }
    }
  }
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VRadioGroup/VRadio.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      const color = (this.radioGroup || {}).validationState
+
+      return this.isActive ? this.color : (color || false)
+    },
+    computedIcon (): string {
+      return this.isActive
+        ? this.onIcon
+        : this.offIcon
+    },
+    hasLabel () {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/VRadio.ts]: `: boolean`

File: packages/vuetify/src/components/VRadioGroup/VRadio.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      if (this.name || !this.radioGroup) {
+        return this.name
+      }
+
+      return this.radioGroup.name || `'v-radio-'${this.radioGroup._uid}`
+    }
+  },
+
+  methods: {
+    genInput (args: any) {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/VRadio.ts]: `: VNode`

File: packages/vuetify/src/components/VRadioGroup/VRadio.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+      // We can't actually use the mixin directly because
+      // it's made for standalone components, but its
+      // genInput method is exactly what we need
+      return Selectable.options.methods.genInput.call(this, 'radio', args)
+    },
+    genLabel () {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/VRadio.ts]: same

File: packages/vuetify/src/components/VRadioGroup/VRadio.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        attrs: {
+          for: this.id
+        },
+        props: {
+          color: ((this.radioGroup || {}).validationState) || false,
+          focused: this.hasState
+        }
+      }, getSlot(this, 'label') || this.label)
+    },
+    genRadio () {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/VRadio.ts]: same

File: packages/vuetify/src/components/VRadioGroup/VRadio.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    onBlur (e: Event) {
+      this.isFocused = false
+      this.$emit('blur', e)
+    },
+    onChange () {
+      if (this.isDisabled || this.isReadonly || this.isActive) return
+
+      this.toggle()
+    },
+    onKeydown () {}
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/VRadio.ts]: maybe remove it?

File: packages/vuetify/src/components/VRadioGroup/VRadioGroup.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    // If no value set on VRadio
+    // will match valueComparator
+    // force default to null
+    value: {
+      default: null
+    }
+  },
+
+  computed: {
+    classes () {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/VRadioGroup.ts]: `: object`

File: packages/vuetify/src/components/VRadioGroup/VRadioGroup.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      return {
+        'v-input--selection-controls v-input--radio-group': true,
+        'v-input--radio-group--column': this.column && !this.row,
+        'v-input--radio-group--row': this.row
+      }
+    }
+  },
+
+  methods: {
+    genDefaultSlot () {
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/VRadioGroup.ts]: `: VNode`

File: packages/vuetify/src/components/VRadioGroup/__tests__/VRadio.spec.ts [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+} from '@vue/test-utils'
+
+describe('VRadio.ts', () => {
+  type Instance = InstanceType<typeof VRadio>
+  let mountFunction: (options?: object) => Wrapper<Instance>
+
+  beforeEach(() => {
+    mountFunction = (options = {}) => {
+      return mount(VRadio, {
+        ...options
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/__tests__/VRadio.spec.ts]: We don't need `...` here. just pass `options`

File: packages/vuetify/src/components/VRadioGroup/__tests__/VRadio.spec.ts [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+  })
+
+  it('should check/uncheck the internal input', () => {
+    const wrapper = mountFunction()
+
+    expect(wrapper.vm.$refs.input.checked).toBe(false)
+
+    wrapper.setData({ isActive: true })
+
+    expect(wrapper.vm.$refs.input.checked).toBe(true)
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/__tests__/VRadio.spec.ts]: `.toBeTruthy`/`.toBeFalsy` please

File: packages/vuetify/src/components/VRadioGroup/__tests__/VRadioGroup.spec.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+// Utilities
+import {
+  mount,
+  Wrapper
+} from '@vue/test-utils'
+
+describe('VRadioGroup.ts', () => {
+  type Instance = InstanceType<typeof VRadioGroup>
+  let mountFunction: (options?: object) => Wrapper<Instance>
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/__tests__/VRadioGroup.spec.ts]: `options?: MountOptions<Instance>`

File: packages/vuetify/src/components/VRadioGroup/__tests__/VRadioGroup.spec.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+  beforeEach(() => {
+    mountFunction = (options = {}) => {
+      return mount(VRadioGroup, {
+        ...options
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/__tests__/VRadioGroup.spec.ts]: Don't need `...`

File: packages/vuetify/src/components/VRadioGroup/__tests__/VRadio.spec.ts [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+// Utilities
+import {
+  mount,
+  Wrapper
+} from '@vue/test-utils'
+
+describe('VRadio.ts', () => {
+  type Instance = InstanceType<typeof VRadio>
+  let mountFunction: (options?: object) => Wrapper<Instance>
```
Comment (sh7dm) [inline packages/vuetify/src/components/VRadioGroup/__tests__/VRadio.spec.ts]: `options?: MountOptions<Instance>`

File: packages/vuetify/src/components/VRadioGroup/VRadioGroup.ts (hunk 7 above)
Comment (johnleider) [reply packages/vuetify/src/components/VRadioGroup/VRadioGroup.ts]: this is inferred

File: packages/vuetify/src/components/VRadioGroup/VRadio.ts (hunk 4 above)
Comment (johnleider) [reply packages/vuetify/src/components/VRadioGroup/VRadio.ts]: this is inferred

File: packages/vuetify/src/components/VRadioGroup/VRadio.ts (hunk 3 above)
Comment (johnleider) [reply packages/vuetify/src/components/VRadioGroup/VRadio.ts]: this is inferred

File: packages/vuetify/src/components/VRadioGroup/VRadio.ts (hunk 2 above)
Comment (johnleider) [reply packages/vuetify/src/components/VRadioGroup/VRadio.ts]: this is inferred

**Pull request merged**